vsphere/vsphere.go

```diff
@@ -755,9 +764,13 @@ func (ops *vsphereOps) getDatastoreToUseInStoragePod(
 	}
 
 	spec.DeviceChange = deviceChange
-	ops.dsLock.LockWithKey(ops.cfg.VMUUID, storagePod.Name())
+	storeLock, lockErr := ops.dsLock.LockWithKey(ops.cfg.VMUUID, storagePod.Name())
 	recommendedDatastore, err := recommendDatastore(ctx, vmObj, storagePod, spec)
-	ops.dsLock.UnlockWithKey(storagePod.Name())
+	// We are doing a best effort
+	if lockErr != nil {
+		ops.dsLock.Unlock(storeLock)
```

Comment:
should we call Unlock? I would not expect storeLock to have a valid value when lockErr is not nil. Perhaps just log a message?